backend/app.py

```diff
@@ -96,6 +96,23 @@ def read_root():
 #
 # cache = memoize(maxsize=1000)
 
+# TODO: Get timeout to work
+import asyncio
+from fastapi import HTTPException
+async def simulate_long_running_task():
+    """Simulate"""
+    await asyncio.sleep(10)
+    return "Computation complete"
+
+@APP.get("/test-timeout")
+async def test_timeout(timeout=3):
+    """Test timeout"""
+    try:
+        result = await asyncio.wait_for(simulate_long_running_task(), timeout=timeout)
```

Comment:
## Working method: Using Python > `gunicorn`/`uvicorn` options
@Sigfried I tried `gunicorn` as well, but those options didn't work either.

I went back to ChatGPT and tried asking the question generally. It came up with this, and it worked the first time!

Let's just go with our middleware option.

I have to complete some of my tickets first, so if you want me to do this, it'll have to be some time in January. But I would be happy if you tackled this too.

Reply:
## How to get this to work for all routes (middleware)
Here's what ChatGPT said: https://chat.openai.com/share/5d1061b3-b114-42fd-a3eb-0b67adad7631